stories/s1grdiw.eocloud.stories.js

```diff
@@ -0,0 +1,203 @@
+
+  const perform = async () => {
+    const layer = (await LayersFactory.makeLayers(`${DATASET_EOCLOUD_S1GRD.shServiceHostname}v1/wms/${instanceId}`, (lId, datasetId) => (layerId === lId)))[0];
+
+    const getMapParams = {
+      bbox: bbox,
+      fromTime: new Date(Date.UTC(2018, 11 - 1, 22, 0, 0, 0)),
+      toTime: new Date(Date.UTC(2018, 12 - 1, 22, 23, 59, 59)),
+      width: 512,
+      height: 512,
+      format: MimeTypes.JPEG,
+    };
+    const imageBlob = await layer.getMap(getMapParams, ApiType.WMS);
+    img.src = URL.createObjectURL(imageBlob);
+  };
+  perform().then(() => {});
+
+  return wrapperEl;
+};
+
+export const getMapWMSEvalscript = () => {
+  const img = document.createElement('img');
+  img.width = '512';
+  img.height = '512';
+
+  const wrapperEl = document.createElement('div');
+  wrapperEl.innerHTML = "<h2>GetMap with WMS - evalscript</h2>";
+  wrapperEl.insertAdjacentElement("beforeend", img);
+
+  const perform = async () => {
+    const layer = new S1GRDEOCloudLayer(
+      instanceId,
+      layerId,
+      `
+        return [2.5 * VV, 1.5 * VV, 0.5 * VV];
+      `,
+      null, null, null, AcquisitionMode.IW, Polarization.DV, Resolution.HIGH
+    );
+
+    const getMapParams = {
+      bbox: bbox,
+      fromTime: new Date(Date.UTC(2018, 11 - 1, 22, 0, 0, 0)),
+      toTime: new Date(Date.UTC(2018, 12 - 1, 22, 23, 59, 59)),
+      width: 512,
+      height: 512,
+      format: MimeTypes.JPEG,
+      maxCCPercent: 100,
+    };
+    const imageBlob = await layer.getMap(getMapParams, ApiType.WMS);
+    img.src = URL.createObjectURL(imageBlob);
+  };
+  perform().then(() => {});
+
+  return wrapperEl;
+};
+
+export const findTiles = () => {
+  const layer = new S1GRDEOCloudLayer(instanceId, layerId, null, null, null, null, AcquisitionMode.IW, Polarization.DV, Resolution.HIGH);
+  const containerEl = document.createElement('pre');
+
+  const wrapperEl = document.createElement('div');
+  wrapperEl.innerHTML = "<h2>findTiles</h2>";
+  wrapperEl.insertAdjacentElement("beforeend", containerEl);
+
+  const perform = async () => {
+    const data = await layer.findTiles(
+      bbox,
+      new Date(Date.UTC(2020, 1 - 1, 1, 0, 0, 0)),
+      new Date(Date.UTC(2020, 1 - 1, 15, 23, 59, 59)),
+      5,
+      null,
+      OrbitDirection.ASCENDING,
+    );
+    renderTilesList(containerEl, data.tiles);
+  };
+  perform().then(() => {});
+
+  return wrapperEl;
+};
+
+function renderTilesList(containerEl, list) {
+  list.forEach(tile => {
+    const ul = document.createElement('ul');
+    containerEl.appendChild(ul);
+    for (let key in tile) {
+      const li = document.createElement('li');
+      ul.appendChild(li);
+      let text;
+      if (tile[key] instanceof Object) {
+        text = JSON.stringify(tile[key]);
+      } else {
+        text = tile[key];
+      }
+      li.innerHTML = `${key} : ${text}`;
+    }
+  });
+}
+
+async function setAuthTokenWithOAuthCredentials () {
```

Comment:
This function probably isn't needed in EOCloud stories. If the authentication is needed, it's probably done in a different way than for the S1 on AWS.

Reply:
Thx, removed.

Reply:
Nice, resolving.